Client/Frontend/Widgets/PhotonActionSheetProtocol.swift

```diff
@@ -190,6 +191,21 @@ extension PhotonActionSheetProtocol {
             }
         }
 
+        let removeTopSitesPin = PhotonActionSheetItem(title: Strings.RemovePinTopsiteActionTitle, iconString: "action_unpin") { action in
+            guard let url = tab.url?.displayURL,
+                let sql = self.profile.history as? SQLiteHistory else { return }
+            let absoluteString = url.absoluteString
+
+            sql.getSitesForURLs([absoluteString]) >>== { result in
+                guard let siteOp = result.asArray().first, let site = siteOp else {
```

Comment:
The double optional on site is weird here. But this works too 
`let site = val.successValue?.asArray().first?.flatMap({ $0 })`